flytekit/configuration/__init__.py

```diff
@@ -5,28 +5,71 @@
 
 .. currentmodule:: flytekit.configuration
 
-Flytekit Configuration Ecosystem
---------------------------------
+Flytekit Configuration Sources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Where can configuration come from?
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+There are multiple ways to configure flytekit settings:
 
-- Command line arguments. This is the ideal location for settings to go. (See ``pyflyte package --help`` for example.)
-- Environment variables. Users can specify these at compile time, but when your task is run, Flyte Propeller will also set configuration to ensure correct interaction with the platform.
-- A config file - an INI style configuration file. By default, flytekit will look for a file in two places
-  1. First, a file named ``flytekit.config`` in the Python interpreter's starting directory
-  2. A file in ``~/.flyte/config`` in the home directory as detected by Python.
+**Command Line Arguments**: This is the recommended way of setting configuration values for many cases.
+For example, see `pyflyte package <pyflyte.html#pyflyte-package>`_ command.
+
+**Python Config Object**: A :py:class:`~flytekit.configuration.Config` object can by used directly, e.g. when
+initializing a :py:class:`~flytefit.remote.remote.FlyteRemote` object. See :doc:`here <design/control_plane>` for examples on
+how to specify a ``Config`` object.
+
+**Environment Variables**: Users can specify these at compile time, but when your task is run, Flyte Propeller will
+also set configuration to ensure correct interaction with the platform. The environment variables must be specified
+with the format ``FLYTE_{SECTION}_{OPTION}``, all in upper case. For example, to specify the
+:py:class:`PlatformConfig.endpoint <flytekit.configuration.PlatformConfig>` setting, the environment variable would
+be ``FLYTE_PLATFORM_URL``.
+
+.. note::
+
+   Environment variables won't work for image configuration, which need to be specified with the
+   `pyflyte package --image ... <pyflyte.html#cmdoption-pyflyte-package-i>`_ option or in a configuration
+   file.
+
+**YAML Format Configuration File**: A configuration file that contains settings for both
+`flytectl <https://docs.flyte.org/projects/flytectl/>`__ and ``flytekit``. This is the recommended configuration
+file format. Invoke ``flytectl --help`` to learn about all of the configuration yaml options.
```

Comment:
should we mention that it might be helpful to generate this file from calling the `flytectl init config` command?  and that the image section, even though it's not respected by flytectl, can still be understood by flytekit?  or will this be confusing?  it feels a bit confusing.